### ISSUE TITLE
Is ext-gmp really needed?

### ISSUE THREAD

**Issue by vbartusevicius** [edamov/pushok#39]
Tried to install this library, but composer complained about missing `ext-gmp` php extension. I quickly looked for `gmp_` in source tree and didn't found anything.

So is it really needed?

**Comment by edamov** [comment]
Hi, this extension is really needed. One of dependency wouldn't work without it

**Comment by xgin** [comment]
`ext-gmp` is actually required only for key conversion.
Thanks to the interface of `Pushok\AuthProvider\Token` class, it is possible to generate a key in one process (e.g. in a docker container):
```
$authProvider = Pushok\AuthProvider\Token::create($options);
echo $authProvider->get() . PHP_EOL;
```
And then reuse it in a production environment without `ext-gmp`:
```
$authProvider = Pushok\AuthProvider\Token::useExisting($token, $options);
```

**Comment by csimpi** [comment]
Please, remove GMP dependency, it's ridiculously huge footprint and nothing is really using it as I see. 

@xgin Did you solve the problem without GMP? We can't use this package because of this dependency and would be great to avoid this issue and still use the lib.

@edamov Still think it's that necessary?

**Comment by xgin** [comment]
@csimpi Yep, with https://github.com/web-token/jwt-framework/pull/179 I managed to use the library without GMP support.

**Comment by edamov** [comment]
@csimpi I agree. But how do you propose to solve this issue? Just remove it from dependency? And what if someone needs to use key conversion and the extension will be required in this case?
Can you create PR?

**Comment by csimpi** [comment]
@edamov Which part / package requires GMP? I haven't dive into the code.

**Comment by edamov** [comment]
@csimpi We need it when we generate token:
``` php 
$authProvider = Pushok\AuthProvider\Token::create($options); 
```
And the library `web-token`, which I use as dependency for generating token, requires GMP dependency.

**Comment by csimpi** [comment]
@edamov I don't see GMP as a requirement of web-token package, maybe they removed it in the meantime? 

Btw, you don't need that package at all, I think you can generate that token manually. I'll try to come back to you with a solution, I'm coding it manually rn.

**Comment by rasmusbe** [comment]
I think that if we don't use `ext-gmp` directly then we shouldn't have it in our requirements.
If `web-token` needs it and has it in their requirements it will be handled by composer correctly anyways since the requirements for `web-token` is checked as well.

**Comment by edamov** [comment]
Fix with new tag 0.8.0 was just created. Thanks to everyone!
